graph/coloring/coloring_test.go

```diff
@@ -801,3 +801,64 @@ func BenchmarkColoring(b *testing.B) {
 		})
 	}
 }
+
+func BenchmarkColoringSets(b *testing.B) {
+	for _, bench := range coloringTests {
+		b.Run(bench.name, func(b *testing.B) {
+			b.Run("Dsatur", func(b *testing.B) {
+				_, colors, err := Dsatur(bench.g, nil)
+				if err != nil {
+					b.Fatalf("coloring failed: %v", err)
+				}
+
+				b.ResetTimer()
+				for i := 0; i < b.N; i++ {
+					_ = Sets(colors)
+				}
+			})
+			if !bench.long || *runLong {
+				b.Run("DsaturExact", func(b *testing.B) {
+					_, colors, err := DsaturExact(nil, bench.g)
+					if err != nil {
+						b.Fatalf("coloring failed: %v", err)
+					}
```

Comment:
This is very expensive for this benchmark. I don't think we need to benchmark each kind of colouring algorithm here, so just use DSATUR. Also, do the set construction outside the benchmark `Run` so that it doesn't need to be rebuilt for each run on the benchmark.

Alternatively, just drop this. I'm not convinced that this is worth benchmarking. It's essentially an output routine.

Reply:
Ah okay, I wasn't aware, thanks for enlightening me.

I admit I'm unclear what a set construction is or which part of this code does it, so I was wondering if you can clarify?

Regardless, I'm more than happy to follow your judgement, so I'll remove the `DsaturExact` case. :)

Reply:
D'oh! I was having a silly moment. I've just realised that I'm benchmarking the `Sets` function, not the construction functions like `SanSegundo` and `Dsatur`! Thanks for clarifying. 😁

Cool, in that case I agree that dropping the benchmarks makes sense. On it.

Reply:
The benchmarks are finally published, so I'll resolve this conversation for now. :)